pkg/grpc/grpc.go

```diff
@@ -160,15 +157,18 @@ func (g *Manager) GetGRPCConnection(ctx context.Context, address, id string, nam
 	}
 
 	opts = append(opts, customOpts...)
+
+	ctx, cancel := context.WithTimeout(parentCtx, dialTimeout)
 	conn, err := grpc.DialContext(ctx, dialPrefix+address, opts...)
+	cancel()
 	if err != nil {
 		return nil, func() {}, err
 	}
 
 	teardown := releaseFactory(conn)
 	g.lock.Lock()
-	defer g.lock.Unlock()
 	g.connectionPool.Register(address, conn)
+	g.lock.Unlock()
```

Comment:
Is there a possibility of flow never reaching this line due to some issue in Register function ?

Reply:
No (aside from perhaps a panic? But that would be a different - and bigger - problem)